File: hdu/1207.cpp
#include<iostream>
#include<cstring>
#include<algorithm>
#include<cstdio>
#include<cmath>
#include<string>
#include<map>
#include<set>
#include<vector>
#include<cstdlib>
#include<queue>
using namespace std;
int solve(int n){
    int cnt=1;
    while(n!=1){
        if(n%2!=0) n=3*n+1;
        else n/=2;
        cnt++;
    }
    return cnt;
}
int main(){
    int x,y;
    while(cin>>x>>y){
        if(x>y){ int t=x;x=y;y=t;}
        int ret=0;
        for(int i=x;i<=y;i++){
            int tmp=solve(i);
            if(ret<tmp) ret=tmp;
        }
        cout<<x<<' '<<y<<' '<<ret<<endl;
    }
    return 0;
}

File: hdu/1207.java
import java.util.*;
import java.io.*;
import java.math.BigInteger;
public class 1207{
	public int solve(int i){
		int cnt=0;
		while(i!=1){
			if(n&1) n=3*n+1;
			else n=n/2;
			cnt++;
		}
		return cnt;
	}
	public static void main(){
		Scanner cin=new Scanner(System.in);
		while(input.hasNext()){
			int x,y;
			x=cin.nextInt();y=cin.nextInt();
			int ret=-1;
			for(int i=x;i<=y;i++){
				int tmp=solve(i);
				if(ret<tmp) ret=tmp;
			}
			System.out.println(ret);
		}
	}
}